### PR TITLE
Fix some things on Travis

### PR DESCRIPTION
**Pull request by jwiggins**
* Now using the latest pip
* `imread` wasn't actually being installed
* Optional package installation uses a requirements file
* `python setup.py develop` instead of the previous two-step.

@blink1073: @emmanuelle tells me you're the resident Travis expert. Please take a look when you have some time.

`pyamg` installation was failing in my test branch. Hopefully it works here :pray:

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,8 +8,7 @@
 language: python
 sudo: false
 cache:
-  directories:
-    - $HOME/.cache/pip
+  - pip
```
Comment (blink1073) [inline .travis.yml]: We needed the former because we override the `install` step.  http://docs.travis-ci.com/user/caching/#pip-cache

Comment (blink1073) [inline .travis.yml]: Perhaps add a comment so we don't keep rediscovering this? :smile:

Comment (jwiggins) [inline .travis.yml]: Ooh. Good to know. I'll leave a note.

**Comment by jwiggins** [comment]
:grimacing: Baby steps. The `pip` upgrade was happening _outside_ the virtualenv... I've also added some `pip list` invocations to give an idea of what's installed.

The `skimage.io.tests.test_imread` tests failed again on 2.7. I'm not sure what's the best way to debug that.

**PS:** #1691 has three builds queued up right now. Two of those should be canceled.

**Comment by jwiggins** [comment]
`imread`... :rage:

@blink1073: **TL;DR**: Rebuild imread for 2.7. It looks like it might be built against a version of numpy which is newer than what is running on Travis.

Here's the message when importing `imread`:
```
RuntimeError: module compiled against API version 9 but this version of numpy is 6
RuntimeError: module compiled against API version 9 but this version of numpy is 6
Could not import submodules (exact error was: numpy.core.multiarray failed to import).
There are many reasons for this error the most common one is that you have
either not built the packages or have built (using `python setup.py build`) or
installed them (using `python setup.py install`) and then proceeded to test
mahotas **without changing the current directory**.
Try installing and then changing to another directory before importing mahotas.
```
`imread` has the fun property that it swallows import errors when it can't load its extension module. See:
https://github.com/luispedro/imread/blob/master/imread/__init__.py#L13-L24
... which means that this:
https://github.com/scikit-image/scikit-image/blob/master/skimage/io/_plugins/imread_plugin.py#L5-L10
won't work as intended. Good times.

**Comment by jwiggins** [comment]
Hey, listing the installed packages turned out to be a good idea :smile:

It appears that the 2.7 tests are getting numpy 1.6.1 and _all_ the others are getting 1.9.2. I'm not quite sure why that's happening.

**Comment by blink1073** [comment]
Aha!  That is on purpose.  Our 2.7 build is against minimum versions of each dependency.

**Comment by jwiggins** [comment]
So does imread need to be rebuilt against numpy 1.6.1?

**Comment by blink1073** [comment]
I'd say we just skip it on 2.7, it is optional.

**Comment by jwiggins** [comment]
Lots of breakage now, but none of it caused by my changes :angel:

**Comment by blink1073** [comment]
Travis is being finicky this past day...

**Comment by jwiggins** [comment]
So... 2.6 keeps timing out; here and on master. Do you think it's a transient problem with Travis or could something actually be broken?

**Comment by blink1073** [comment]
Not sure, I opened #1692 for discussion.

**Comment by blink1073** [comment]
I'm calling this one good, thanks @jwiggins!

**Comment by jwiggins** [comment]
:tada: Next up, ccache...

**Pull request merged**